### PR TITLE
Qinusty/bb-asset-hub Code Drop

### PR DESCRIPTION
**Pull request by Qinusty**
This PR contains a series of commits from the [qinusty/bb-asset-hub](https://github.com/Qinusty/bb-asset-hub) project at [01b36da](https://github.com/Qinusty/bb-asset-hub/commit/01b36da099706fedbff99437d396b2983f19edb8).

This merges the work into the buildbarn namespace for continued development.

### CODE REVIEW

File: cmd/bb_remote_asset/main.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+	bb_digest "github.com/buildbarn/bb-storage/pkg/digest"
+	"github.com/buildbarn/bb-storage/pkg/global"
+	bb_grpc "github.com/buildbarn/bb-storage/pkg/grpc"
+	"github.com/buildbarn/bb-storage/pkg/util"
+	"github.com/gorilla/mux"
+
+	"google.golang.org/grpc"
+)
+
+const (
```
Comment (EdSchouten) [inline cmd/bb_remote_asset/main.go:25]: I think this is a leftover from bb-browser code that can be removed.

File: internal/mock/aliases/aliases.go [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+// Go standard library. The only reason this file exists is to allow the
+// gomock() Bazel rule to emit mocks for them, as that rule is only
+// capable of emitting mocks for interfaces built through a
+// go_library().
+
+// ReadCloser is an alias of io.ReadCloser.
+type ReadCloser = io.ReadCloser
+
+// Writer is an alias of io.Writer.
+type Writer = io.Writer
```
Comment (EdSchouten) [inline internal/mock/aliases/aliases.go:17]: Nit: Writer isn't used?

File: tools/github_workflows/workflows_template.libsonnet [hunk 3]
```diff
@@ -0,0 +1,141 @@
+{
```
Comment (EdSchouten) [inline tools/github_workflows/workflows_template.libsonnet:1]: Why has this been copied? Would it be possible to remotely reference the workflow template we have in bb-storage?

Comment (tomcoldrick-ct) [reply tools/github_workflows/workflows_template.libsonnet]: Purely an oversight on my part, removed now.

**Review by EdSchouten — APPROVED** [review]

**Pull request merged**